arch/arm/src/s32k3xx/hardware/s32k3xx_emac.h

```diff
@@ -0,0 +1,3084 @@
+#define EMAC_MAC_PACKET_FILTER_SAIF       (1 << 8)  /* Bit 8: SA Inverse Filtering */
+#define EMAC_MAC_PACKET_FILTER_SAF        (1 << 9)  /* Bit 9: Source Address Filter Enable */
+#define EMAC_MAC_PACKET_FILTER_HPF        (1 << 10) /* Bit 10: Hash Or Perfect Filter */
+#define EMAC_MAC_PACKET_FILTER_VTFE       (1 << 16) /* Bit 16: VLAN Tag Filter Enable */
+#define EMAC_MAC_PACKET_FILTER_IPFE       (1 << 20) /* Bit 20: Layer 3 and Layer 4 Filter Enable */
+#define EMAC_MAC_PACKET_FILTER_DNTU       (1 << 21) /* Bit 21: Drop Non-TCP/UDP Over IP Packets */
+#define EMAC_MAC_PACKET_FILTER_RA         (1 << 31) /* Bit 31: Receive All */
+
+/* MAC Watchdog Timeout (MAC_WATCHDOG_TIMEOUT) */
+#define EMAC_MAC_WATCHDOG_TIMEOUT_WTO_SHIFT  (0) /* Bits 0-4: Watchdog Timeout */
+#define EMAC_MAC_WATCHDOG_TIMEOUT_WTO_MASK   (0xF << EMAC_MAC_WATCHDOG_TIMEOUT_WTO_SHIFT)
+#define EMAC_MAC_WATCHDOG_TIMEOUT_WTO(n)     ((n << EMAC_MAC_WATCHDOG_TIMEOUT_WTO_SHIFT) & EMAC_MAC_WATCHDOG_TIMEOUT_WTO_MASK)
+#define EMAC_MAC_WATCHDOG_TIMEOUT_PWE        (1 << 8) /* Bit 8: Programmable Watchdog Enable */
+
+/* MAC Hash Table First 32 Bits (MAC_HASH_TABLE_REG0) */
+#define EMAC_MAC_HASH_TABLE_REG0_HT31T0_SHIFT  (0) /* Bits 0-32: MAC Hash Table First 32 Bits */
+#define EMAC_MAC_HASH_TABLE_REG0_HT31T0_MASK   (0xFFFFFFFF << EMAC_MAC_HASH_TABLE_REG0_HT31T0_SHIFT)
+#define EMAC_MAC_HASH_TABLE_REG0_HT31T0(n)     ((n << EMAC_MAC_HASH_TABLE_REG0_HT31T0_SHIFT) & EMAC_MAC_HASH_TABLE_REG0_HT31T0_MASK)
+
+/* MAC Hash Table Second 32 Bits (MAC_HASH_TABLE_REG1) */
+#define EMAC_MAC_HASH_TABLE_REG1_HT63T32_SHIFT  (0) /* Bits 0-32: MAC Hash Table Second 32 Bits */
+#define EMAC_MAC_HASH_TABLE_REG1_HT63T32_MASK   (0xFFFFFFFF << EMAC_MAC_HASH_TABLE_REG1_HT63T32_SHIFT)
+#define EMAC_MAC_HASH_TABLE_REG1_HT63T32(n)     ((n << EMAC_MAC_HASH_TABLE_REG1_HT63T32_SHIFT) & EMAC_MAC_HASH_TABLE_REG1_HT63T32_MASK)
+
+/* MAC VLAN Tag (MAC_VLAN_TAG) */
+#define EMAC_MAC_VLAN_TAG_VL_SHIFT     (0) /* Bits 0-16: VLAN Tag Identifier for Receive Packets */
+#define EMAC_MAC_VLAN_TAG_VL_MASK      (0xFFFF << EMAC_MAC_VLAN_TAG_VL_SHIFT)
+#define EMAC_MAC_VLAN_TAG_VL(n)        ((n << EMAC_MAC_VLAN_TAG_VL_SHIFT) & EMAC_MAC_VLAN_TAG_VL_MASK)
+#define EMAC_MAC_VLAN_TAG_ETV          (1 << 16) /* Bit 16: Enable Tag For VLAN */
+#define EMAC_MAC_VLAN_TAG_VTIM         (1 << 17) /* Bit 17: VLAN Tag Inverse Match Enable */
+#define EMAC_MAC_VLAN_TAG_ESVL         (1 << 18) /* Bit 18: Enable S-VLAN */
+#define EMAC_MAC_VLAN_TAG_ERSVLM       (1 << 19) /* Bit 19: Enable Receive S-VLAN Match */
+#define EMAC_MAC_VLAN_TAG_DOVLTC       (1 << 20) /* Bit 20: Disable VLAN Type Check */
+#define EMAC_MAC_VLAN_TAG_EVLS_SHIFT   (21)      /* Bits 21-23: Enable VLAN Tag Stripping */
+#define EMAC_MAC_VLAN_TAG_EVLS_MASK    (0x3 << EMAC_MAC_VLAN_TAG_EVLS_SHIFT)
+#define EMAC_MAC_VLAN_TAG_EVLS(n)      ((n << EMAC_MAC_VLAN_TAG_EVLS_SHIFT) & EMAC_MAC_VLAN_TAG_EVLS_MASK)
```

Comment:
```suggestion
#define EMAC_MAC_VLAN_TAG_EVLS(n)      (((n) << EMAC_MAC_VLAN_TAG_EVLS_SHIFT) & EMAC_MAC_VLAN_TAG_EVLS_MASK)
```
here and other places